### PR TITLE
Identification of Alzheimer's Disease in Brain Scans Using a Siamese Neural Network

### PR DESCRIPTION
**Pull request by algo-rizz-em**
Hi teaching team,

This is a pull request of my solution for project number 7: create a classifier based on Siamese network to classify Alzheimer’s disease (normal and AD) of the ADNI brain data set. I was able to achieve a typical accuracy of around 77%, with around 2% of variance over and under that in different runs. 

Please refer to the README file for more information about my implementation.

Thank you.

Minhao Sun, 46417259

### CODE REVIEW

**Comment by algo-rizz-em** [comment]
Addendum:
The project contains the following items:
- `dataset.py`: contains code for preprocessing and loading data
- `modules.py`: contains the components of the models used in the project
- `predict.py`: contains code demonstrating example usage of a trained model
- `train.py`: contains code for training, validating and testing the models
- `utils.py`: contains code for loading and saving the models
- `CONSTANTS.py`: defines global constants for file paths relevant to the project
- `BSD_new.txt`: see the section ***Open Source Acknowledgement and Licences*** in the README
- `README.md`: the README file
- `/assets/`: folder of images used in the README

This project is developed and tested with the following dependencies:
* python 3.11.4 
* pytorch 2.0.1
* torchvision 0.15.2
* numpy 1.25.0
* matplotlib 3.7.1

Please refer to the README file for more information about my implementation.

Cheers,
Minhao

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Very good
- Clear flow of information
- detailed information provided on model architecture, experiments and usage
- relevant plots for loss and accuracy provided
- visualizations shown for predictions for both siamese and classifier
- results are followed by a discussion/interpretation

Commit messages: Good

Code: 
- good design
- sufficiently commented
- sufficient hyperparameter tuning and feature extractor variations tested

Functionality/Performance: 
- both siamese and classifier seem to have overfit, and the classifier achieves 77% for test set

General comments:
- algorithm solves the problem apporpriately with correct network architectures

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
